src/memory-analyzer/gdb_api.h

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************\
+
+Module: GDB Machine Interface API
+
+Author: Malte Mues <mail.mues@gmail.com>
+        Daniel Poetzl
+
+\*******************************************************************/
+
+/// \file
+/// Low-level interface to gdb
+///
+/// This provides an API to run a program under gdb up to some
+/// breakpoint, and then query the values of expressions. It uses the
+/// gdb machine interface (see section "The GDB/MI Interface" in the
+/// gdb manual to communicate with gdb.
+
+// clang-format off
+#if defined(__linux__) || \
+    defined(__FreeBSD_kernel__) || \
+    defined(__GNU__) || \
+    defined(__unix__) || \
+    defined(__CYGWIN__) || \
+    defined(__MACH__)
```

Comment:
should this just emit an error if this is false? Surely it's an error to include this header if it's API can't be implemented